### ISSUE TITLE
FastBoot broken without jQuery

### ISSUE THREAD

**Issue by simonihmig** [san650/ember-cli-page-object#397]
After removing jQuery, I get this in a app when it runs in FastBoot:

```
/Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor.js:119501
	var el = document.createElement("fieldset");
	                  ^

TypeError: Cannot read property 'createElement' of undefined
    at assert (/Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor/ecpo-jquery/dist/jquery.js:882:1)
    at /Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor/ecpo-jquery/dist/jquery.js:2701:1
    at /Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor/ecpo-jquery/dist/jquery.js:2752:1
    at /Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor/ecpo-jquery/dist/jquery.js:36:1
    at /Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor/ecpo-jquery/dist/jquery.js:40:1
    at ContextifyScript.Script.runInContext (vm.js:59:29)
    at VMSandbox.eval (/Users/simonihmig/Projects/tlw/node_modules/fastboot/src/vm-sandbox.js:14:16)
    at /Users/simonihmig/Projects/tlw/node_modules/fastboot/src/ember-app.js:156:15
   ...
```

When jQuery is missing, the own version shipping with this addon is imported here: 
https://github.com/san650/ember-cli-page-object/blob/17dbadd4e23310a9948a0767bff6ba395fc9c9ca/index.js#L28-L31

I am not sure why this is breaking FastBoot, while when Ember CLI imports jQuery (when it is not excluded by setting `vendorFiles`), this is not the case!? 🤔

On the other hand, I'm also wondering while jQuery is imported to `vendor.js`, while it is used only for tests. I think it could go into `test-support.js` instead, right?

I quickly tried this out, by adding `, { type: 'test' }` to the `import()` calls. And indeed this fixing the problem (as `test-support.js` is not used in FastBoot), and tests are still running fine. I can easily submit a PR if that seems fine!

The same applies not only to jQuery, but all of this addon's code: couldn't that go into `test-support.js` as well instead of `vendor.js`? There is no point in having the addon's code availabe in dev mode, only in test, or do I miss something. Changing that would be easy as well I think...

**Comment by san650** [comment]
Thanks @simonihmig for reporting this.

> couldn't that go into test-support.js as well instead of vendor.js?

If it's backwards compatible, yes!

I guess a long time ago we moved everything to `addon` folder to change how to import the page object helpers from your application. If I remember correctly, `test-support/` folder was merged with application's `test/` folder making all addon's modules to "magically" appear under app's context. e.g. `import PageObject from "./tests/";`. We wanted to do the imports more explicit `import PageObject from "ember-cli-page-object";`.

I guess we can use `addon-test-support` folder if we change how it's exposed to the application. By default it does this.

> `addon-test-support/` - merged with the application’s `tests/` under the addon’s namespace at `${addon-name}/test-support/`.

https://ember-cli.com/extending/#addon-project-structure
